app/phone/src/main/java/dev/jdtech/jellyfin/fragments/MovieFragment.kt

```diff
@@ -192,72 +192,82 @@ class MovieFragment : Fragment() {
-                if (requireContext().getExternalFilesDirs(null).filterNotNull().size > 1) {
-                    val storageDialog = getStorageSelectionDialog(
-                        requireContext(),
-                        onItemSelected = { storageIndex ->
-                            if (viewModel.item.sources.size > 1) {
-                                val dialog = getVideoVersionDialog(
-                                    requireContext(),
-                                    viewModel.item,
-                                    onItemSelected = { sourceIndex ->
-                                        createDownloadPreparingDialog()
-                                        viewModel.download(sourceIndex, storageIndex)
-                                    },
-                                    onCancel = {
-                                        binding.itemActions.progressDownload.isVisible = false
-                                        binding.itemActions.downloadButton.setIconResource(CoreR.drawable.ic_download)
-                                    },
-                                )
-                                dialog.show()
-                                return@getStorageSelectionDialog
-                            }
-                            createDownloadPreparingDialog()
-                            viewModel.download(storageIndex = storageIndex)
-                        },
-                        onCancel = {
-                            binding.itemActions.progressDownload.isVisible = false
-                            binding.itemActions.downloadButton.setIconResource(CoreR.drawable.ic_download)
-                        },
-                    )
-                    storageDialog.show()
-                    return@setOnClickListener
-                }
-                if (viewModel.item.sources.size > 1) {
-                    val dialog = getVideoVersionDialog(
-                        requireContext(),
-                        viewModel.item,
-                        onItemSelected = { sourceIndex ->
-                            createDownloadPreparingDialog()
-                            viewModel.download(sourceIndex)
-                        },
-                        onCancel = {
-                            binding.itemActions.progressDownload.isVisible = false
-                            binding.itemActions.downloadButton.setIconResource(CoreR.drawable.ic_download)
-                        },
-                    )
-                    dialog.show()
-                    return@setOnClickListener
-                }
-                createDownloadPreparingDialog()
-                viewModel.download()
-            }
+                handleDownload()
         }
 
         binding.peopleRecyclerView.adapter = PersonListAdapter { person ->
             navigateToPersonDetail(person.id)
         }
     }
 
+    private fun handleDownload() {
```

Comment:
A lot of this code is the same between the `EpisodeBottomSheetFragment` and `MovieFragment`. Since a lot of it is UI logic I'm not sure it could be avoided, but I think it'd be worth checking if there's a way to reduce the duplication.

Reply:
Those are 2 different UI fragments might be by design to separate the two when adjustment is needed so I left it as is. I will mark it as resolved.